From 8e71802824b424d12356d32884bc055ac3bd3130 Mon Sep 17 00:00:00 2001
From: Jianhua Xie <Jianhua.Xie@freescale.com>
Date: Fri, 24 Apr 2015 11:27:28 +0800
Subject: [PATCH 45/57] powerpc/t10xx/dts: Remove software feature node from
 platform dt

Configurable node based on software feature should not appear in
platform dt, since platform dt should only describe hardware.  If
some software features depend on some kinds of hardware, those
software should share those hardware, and provide configurable
capabilities in run-time.  At least, a separate device tree with
usdpaa style for software features will instead to keep platform
dt clear.
This patch removes software feature related offline port nodes
and buffer pool nodes from t10xx platform dt.

Change-Id: Ib367f9facfdd6743a704f863728a7f2e0e7955ce
Suggested-by: Pan Jiafei <Jiafei.Pan@freescale.com>
Suggested-by: Madalin Bucur <madalin.bucur@freescale.com>
Suggested-by: Scott Wood <scottwood@freescale.com>
Signed-off-by: Jianhua Xie <jianhua.xie@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/35429
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Madalin-Cristian Bucur <madalin.bucur@freescale.com>
[Original patch taken from QorIQ-SDK-V1.8-20150619-yocto.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/boot/dts/t1023rdb.dts   |   18 ------------------
 arch/powerpc/boot/dts/t1024rdb.dts   |   18 ------------------
 arch/powerpc/boot/dts/t102xqds.dtsi  |   17 -----------------
 arch/powerpc/boot/dts/t1040d4rdb.dts |   18 ------------------
 arch/powerpc/boot/dts/t1040rdb.dts   |   18 ------------------
 arch/powerpc/boot/dts/t104xqds.dtsi  |   17 -----------------
 6 files changed, 0 insertions(+), 106 deletions(-)

diff --git a/arch/powerpc/boot/dts/t1023rdb.dts b/arch/powerpc/boot/dts/t1023rdb.dts
index 333be15..0fd8dc8 100644
--- a/arch/powerpc/boot/dts/t1023rdb.dts
+++ b/arch/powerpc/boot/dts/t1023rdb.dts
@@ -153,14 +153,6 @@
 		};
 	};
 
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1023-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
-
 	fsl,dpaa {
 		compatible = "fsl,t1023-dpaa", "fsl,dpaa";
 		ethernet@0 {
@@ -179,16 +171,6 @@
 			compatible = "fsl,t1023-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm1mac4>;
 		};
-		/* Enable one offline port as default to support HW based LAG */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	pci0: pcie@ffe240000 {
diff --git a/arch/powerpc/boot/dts/t1024rdb.dts b/arch/powerpc/boot/dts/t1024rdb.dts
index 8ed7f11..9bbab68 100644
--- a/arch/powerpc/boot/dts/t1024rdb.dts
+++ b/arch/powerpc/boot/dts/t1024rdb.dts
@@ -186,14 +186,6 @@
 		};
 	};
 
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1024-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
-
 	fsl,dpaa {
 		compatible = "fsl,t1024-dpaa", "fsl,dpaa";
 		ethernet@0 {
@@ -212,16 +204,6 @@
 			compatible = "fsl,t1024-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm1mac4>;
 		};
-		/* Enable one offline port as default to support HW based LAG */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	qe: qe@ffe139999 {
diff --git a/arch/powerpc/boot/dts/t102xqds.dtsi b/arch/powerpc/boot/dts/t102xqds.dtsi
index 03882ca..ab7980d 100644
--- a/arch/powerpc/boot/dts/t102xqds.dtsi
+++ b/arch/powerpc/boot/dts/t102xqds.dtsi
@@ -364,13 +364,6 @@
 			};
 		};
 	};
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1024-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
 
 	fsl,dpaa {
 		compatible = "fsl,t1024-dpaa", "fsl,dpaa";
@@ -390,16 +383,6 @@
 			compatible = "fsl,t1024-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&fm1mac4>;
 		};
-		/* Enable one offline port as default to support HW based LAG */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	pci0: pcie@ffe240000 {
diff --git a/arch/powerpc/boot/dts/t1040d4rdb.dts b/arch/powerpc/boot/dts/t1040d4rdb.dts
index 4725e17..666716f 100644
--- a/arch/powerpc/boot/dts/t1040d4rdb.dts
+++ b/arch/powerpc/boot/dts/t1040d4rdb.dts
@@ -188,14 +188,6 @@
 		};
 	};
 
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1040-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
-
 	fsl,dpaa {
 		compatible = "fsl,t1040-dpaa", "fsl,dpaa";
 		ethernet@0 {
@@ -218,16 +210,6 @@
 			compatible = "fsl,t1040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&enet4>;
 		};
-		/* enable one offline port */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	qe: qe@ffe139999 {
diff --git a/arch/powerpc/boot/dts/t1040rdb.dts b/arch/powerpc/boot/dts/t1040rdb.dts
index eb0c649..4a3f207 100644
--- a/arch/powerpc/boot/dts/t1040rdb.dts
+++ b/arch/powerpc/boot/dts/t1040rdb.dts
@@ -334,14 +334,6 @@
 		};
 	};
 
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1040-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
-
 	fsl,dpaa {
 		compatible = "fsl,t1040-dpaa", "fsl,dpaa";
 		ethernet@0 {
@@ -364,16 +356,6 @@
 			compatible = "fsl,t1040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&enet4>;
 		};
-		/* enable one offline port */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	qe: qe@ffe139999 {
diff --git a/arch/powerpc/boot/dts/t104xqds.dtsi b/arch/powerpc/boot/dts/t104xqds.dtsi
index 69dff11..7474d19 100644
--- a/arch/powerpc/boot/dts/t104xqds.dtsi
+++ b/arch/powerpc/boot/dts/t104xqds.dtsi
@@ -336,13 +336,6 @@
 			};
 		};
 	};
-	/* bp dts definition is borrowed from other USDPAA dts */
-	bp6: buffer-pool@6 {
-		compatible = "fsl,t1040-bpool", "fsl,bpool";
-		fsl,bpid = <6>;
-		fsl,bpool-ethernet-cfg = <0 0 0 1728 0 0xfeedabba>;
-		fsl,bpool-thresholds = <0x100 0x300 0x0 0x0>;
-	};
 
 	fsl,dpaa {
 		compatible = "fsl,t1040-dpaa", "fsl,dpaa";
@@ -366,16 +359,6 @@
 			compatible = "fsl,t1040-dpa-ethernet", "fsl,dpa-ethernet";
 			fsl,fman-mac = <&enet4>;
 		};
-		/* Enable one offline port as default to support HW based LAG */
-		dpa-fman0-oh@2 {
-			compatible = "fsl,dpa-oh";
-			/* Define frame queues for the OH port*/
-			/* <OH Rx error, OH Rx default> */
-			fsl,qman-frame-queues-oh = <0x68 1 0x69 1>;
-			fsl,bman-buffer-pools = <&bp6>;
-			fsl,qman-frame-queues-tx = <0x90 8>;
-			fsl,fman-oh-port = <&fman0_oh2>;
-		};
 	};
 
 	pci0: pcie@ffe240000 {
-- 
1.7.5.4

